### PR TITLE
Add a common handler for both package and symbol orchestrator

### PR DESCRIPTION
**Pull request by joelverhagen**
In response to https://github.com/NuGet/NuGet.Jobs/pull/779#discussion_r306474352.
Depends on https://github.com/NuGet/NuGet.Jobs/pull/779.
Related to https://github.com/NuGet/NuGetGallery/issues/7354.

To verify this, I would recommend checking out the first commit of this PR and comparing the base to the symbol and package handler _before_ I use the base class. I left the unit tests unmerged to avoid regressions.

### CODE REVIEW

File: src/NuGet.Services.Validation.Orchestrator/BaseValidationMessageHandler.cs [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+            _entityService = entityService ?? throw new ArgumentNullException(nameof(entityService));
+            _validationSetProvider = validationSetProvider ?? throw new ArgumentNullException(nameof(validationSetProvider));
+            _validationSetProcessor = validationSetProcessor ?? throw new ArgumentNullException(nameof(validationSetProcessor));
+            _validationOutcomeProcessor = validationOutcomeProcessor ?? throw new ArgumentNullException(nameof(validationOutcomeProcessor));
+            _telemetryService = telemetryService ?? throw new ArgumentNullException(nameof(telemetryService));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        protected abstract ValidatingType ValidatingType { get; }
+        protected abstract bool ShouldNoOpDueToDeletion { get; }
```
Comment (cristinamanum) [inline src/NuGet.Services.Validation.Orchestrator/BaseValidationMessageHandler.cs]: I could not find any scenario for symbols case to not follow the same flow as packages. We can remove this method.

Comment (joelverhagen) [reply src/NuGet.Services.Validation.Orchestrator/BaseValidationMessageHandler.cs]: Alright, awesome!

Comment (joelverhagen) [reply src/NuGet.Services.Validation.Orchestrator/BaseValidationMessageHandler.cs]: Filed an issue here. I would rather do this part separately. The goal of this PR is a pure refactor with no behavior change.
https://github.com/NuGet/Engineering/issues/2602

File: src/NuGet.Services.Validation.Orchestrator/BaseValidationMessageHandler.cs [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using NuGet.Services.Entities;
+using NuGet.Services.ServiceBus;
+using NuGet.Services.Validation.Orchestrator.Telemetry;
+
+namespace NuGet.Services.Validation.Orchestrator
+{
+    public abstract class BaseValidationMessageHandler<T> : IMessageHandler<PackageValidationMessageData> where T : class, IEntity
```
Comment (agr) [inline src/NuGet.Services.Validation.Orchestrator/BaseValidationMessageHandler.cs]: Can we have something more descriptive than `T`?

Comment (joelverhagen) [reply src/NuGet.Services.Validation.Orchestrator/BaseValidationMessageHandler.cs]: I made it `TEntity`

**Comment by joelverhagen** [comment]
Merged with 5b31e1a4bee88c424edd04e54fd092a61e1329c3.

**Pull request closed without merging**